core/thread/inc/ROOT/RConcurrentHashColl.hxx

```diff
@@ -20,12 +21,15 @@ class TRWSpinLock;
 
 namespace Internal {
 
-struct RHashSet;
+struct RHashMap;
+using RUidColl = std::vector<Int_t>;
 
-/// This class is a TS set of unsigned set
+/// This class is a thread-safe associative collection connecting
+/// a 256 bits digest/hash to a collection of uid (integer)
+/// This is used in the handling of the StreamerInfo record in TFile.
```

Comment:
This doesn't seem to belong to `tread/` - it's not a threading utility, it's a utility for StreamerInfo. Consider moving to `io/io`?

Reply:
Until this PR it was used only in multi-thread environment (hence the original location).

Reply:
That's an excellent historical motivation but not a reason to stay this way today :-)